app/store/middleware.js

```diff
@@ -0,0 +1,201 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import DeviceInfo from 'react-native-device-info';
+
+import {ViewTypes} from 'app/constants';
+import Config from 'assets/config';
+
+export function messageRetention() {
+    return (next) => (action) => {
+        if (action.type === 'persist/REHYDRATE') {
```

Comment:
This doesn't take much time, does it? I'm a bit leery about adding more stuff when the app opens, when that's one of our performance bottlenecks

Reply:
on dev mode is super fast, about 300ms

Reply:
That seems a bit long, but I imagine we have bigger stuff to deal with